File: MaxConsecutiveOnes3.py
# Time Complexity : O(N)
# Space Complexity : O(1)
# Did this code successfully run on Leetcode : Yes
# Any problem you faced while coding this : No

# Your code here along with comments explaining your approach
# Using Sliding Window Approach


class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        if not nums:
            return None

        i = 0
        maxlength = 0
        for j in range(len(nums)):
            if nums[j] == 0:
                k -= 1
            if k < 0:
                if nums[i] == 0:
                    k += 1
                i += 1
            else:
                maxlength = max(maxlength, j - i + 1)
        return maxlength

File: Max_Consecutive_ones_3.java
// Time Complexity : O(n) 
// Space Complexity : O(1) 
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No

class Solution {
    public int longestOnes(int[] nums, int k) {
        if(nums.length == 0 || nums == null){
            return 0;
        }

        int start = 0;
        for(int i = 0; i< nums.length; i++){
            int n = nums[i];
            if(n == 0){
                k--;
            }
            if(k<0){
                if(nums[start]==0){
                    k++;
                }
                start++;
            }
        }
        return (nums.length - start);
    }
}